flow/envs/green_wave_env.py

```diff
@@ -433,18 +433,79 @@ def _reroute_if_final_edge(self, veh_id):
                 pos="0",
                 speed="max")
 
-    # FIXME it doesn't make sense to pass a list of edges since the function
-    # returns a flattened list with no padding, so we would lose information
-    def k_closest_to_intersection(self, edges, k):
-        """Return the vehicle IDs of k closest vehicles to an intersection.
+    def k_closest_to_intersection(self, edges, k, padding=False):
+        """Return the vehicle IDs of the k vehicles closest to an intersection.
 
         For each edge in edges, return the ids (veh_id) of the k vehicles
         in edge that are closest to an intersection (the intersection they
         are heading towards).
 
-        - Performs no check on whether or not edge is going towards an
-          intersection or not.
-        - Does no padding if there are less than k vehicles on an edge.
+        This function performs no check on whether or not edges are going
+        towards an intersection or not, it just gets the vehicles that are
+        closest to the end of their edges.
+
+        If there are less than k vehicles on an edge, the function performs
+        padding by adding empty strings "" instead of vehicle ids if the
+        padding parameter is set to True
+
+        Parameters
+        ----------
+        edges : str | str list
+            id of an edge or list of edge ids
+        k : int (> 0)
```

Comment:
Consider renaming k --> num_neighbors or num_nearest for a more descriptive variable name.

Reply:
Done!